python/ray/autoscaler/_private/command_runner.py

```diff
@@ -858,6 +858,14 @@ def run_init(self, *, as_head, file_mounts, sync_run_yet):
                                 self.ssh_command_runner.cluster_name), mount),
                         container=self.container_name,
                         dst=self._docker_expand_user(mount)))
+                try:
+                    self.run(f"[[ $(id -u) = 0 ]] || "
+                             "sudo chown $(id -u):$(id -g) {mount}")
+                except Exception:
+                    cli_logger.warning(
+                        "The running user is neither `root`, "
+                        "nor is sudo installed. Ray may fail to autoscale "
+                        "because of permission issues.")
```

Comment:
give the user something actionable todo to solve this, for example, how to install sudo...

Reply:
(I agree actionable error messages are good, but non-sudo containers are generally intentional)

Reply:
But to Ameer's point, I think this error message should provide more context about why root/sudo is important (The files {X, Y} have a different owner (uid) than the current ssh user (uid) and therefore cannot be used by Ray).

 I think another option here is to call out that they could try calling `chown` on X,Y files in setup commands.

Reply:
> (I agree actionable error messages are good, but non-sudo containers are generally intentional)

Non-`sudo` containers are more common, but that is because most containers just allow the `root` user.